python/mxnet/module/base_module.py

```diff
@@ -244,7 +247,7 @@ def predict(self, eval_data, num_batch=None, merge_batches=True, reset=True,
                 break
             self.forward(eval_batch, is_train=False)
             pad = eval_batch.pad
-            outputs = [out[0:out.shape[0]-pad].asnumpy() for out in self.get_outputs()]
+            outputs = [out[0:out.shape[0]-pad] for out in self.get_outputs()]
```

Comment:
I think in this particular case it is better to keep a relatively unified interface than to keep it (exactly) backward compatible. Otherwise, it is also a question that whether you want to return np.array or ndarray for `iter_predict`. I think since the module API is already not exactly compatible with `FeedForward`, my personal opinion is to use this chance to have a cleaner interface here. I guess it might also perform better without those blocking `asnumpy()` call after each `forward`.